src/Algebra/Graph/Undirected.hs

```diff
@@ -0,0 +1,913 @@
+-----------------------------------------------------------------------------
+-- |
+-- Module     : Algebra.Graph.Undirected
+-- Copyright  : (c) Andrey Mokhov 2016-2019
+-- License    : MIT (see the file LICENSE)
+-- Maintainer : andrey.mokhov@gmail.com
+-- Stability  : experimental
+--
+-- __Alga__ is a library for algebraic construction and manipulation of graphs
+-- in Haskell. See <https://github.com/snowleopard/alga-paper this paper> for the
+-- motivation behind the library, the underlying theory, and implementation details.
+--
+-- This module defines and abstraction over the core data type 'Graph' and associated algorithms.
+-- Undirected graphs satisfy all laws of the
+-- 'Algebra.Graph.Class.Undirected' type class, including the commutativity
+-- of 'connect'.
+-- 'Graph' is an instance of type classes defined in
+-- modules "Algebra.Graph.Class" and "Algebra.Graph.HigherKinded.Class", which
+-- can be used for polymorphic graph construction and manipulation.
+--
+-----------------------------------------------------------------------------
+module Algebra.Graph.Undirected (
+    -- * Algebraic data type for graphs
+    Graph, toGraph,
+
+    -- * Basic graph construction primitives
+    empty, vertex, edge, overlay, connect, vertices, edges, overlays, connects,
+
+    -- * Graph folding
+    foldg,
+
+    -- * Relations on graphs
+    isSubgraphOf,
+
+    -- * Graph properties
+    isEmpty, size, hasVertex, hasEdge, vertexCount, edgeCount, vertexList,
+    edgeList, vertexSet, edgeSet, adjacencyList, neighbours,
+
+    -- * Standard families of graphs
+    path, circuit, clique, biclique, star, stars, tree, forest,
+
+    -- * Graph transformation
+    removeVertex, removeEdge, replaceVertex, mergeVertices,
+    induce, induceJust, toSymmetricRelation,
+
+    -- * Miscellaneous
+    consistent
+
+    ) where
+
+import Control.Applicative (Alternative)
+import Control.DeepSeq
+import Control.Monad (MonadPlus (..))
+import Data.Maybe (fromMaybe)
+import Data.Coerce
+import GHC.Generics
+
+import qualified Algebra.Graph                 as G
+import Algebra.Graph.Internal 
+
+import qualified Algebra.Graph.Relation.Symmetric    as SR
+import qualified Control.Applicative           as Ap
+import qualified Data.IntSet                   as IntSet
+import qualified Data.Set                      as Set
+import qualified Data.Tree                     as Tree
+
+{-| The Undirected 'Graph' data type is an abstraction over the 'Graph' data
+   type and provides the same graph construction
+primitives 'empty', 'vertex', 'overlay' and 'connect'. We define the same 'Num' 
+as 'Graph' instance as a convenient notation for working with graphs:
+
+    > 0           == Vertex 0
+    > 1 + 2       == Overlay (Vertex 1) (Vertex 2)
+    > 1 * 2       == Connect (Vertex 1) (Vertex 2)
+    > 1 + 2 * 3   == Overlay (Vertex 1) (Connect (Vertex 2) (Vertex 3))
+    > 1 * (2 + 3) == Connect (Vertex 1) (Overlay (Vertex 2) (Vertex 3))
```

Comment:
The above list is no longer correct since constructors are not exported -- please use the expressions that are printed by the `Show` instance.

Reply:
Yes! I forgot to change this particular text when I changed the show intance! Thanks for pointing it out!